diff --git a/node_modules/fhenix-confidential-contracts/contracts/FHERC20.sol b/node_modules/fhenix-confidential-contracts/contracts/FHERC20.sol
index 95b21ab..d1c2ea7 100644
--- a/node_modules/fhenix-confidential-contracts/contracts/FHERC20.sol
+++ b/node_modules/fhenix-confidential-contracts/contracts/FHERC20.sol
@@ -440,7 +440,7 @@ abstract contract FHERC20 is IFHERC20, IFHERC20Errors, Context {
         FHE.allowThis(_encTotalSupply);
 
         emit Transfer(from, to, _indicatorTick);
-        emit ConfidentialTransfer(from, to, euint64.unwrap(transferred));
+        emit ConfidentialTransfer(from, to, bytes32(euint64.unwrap(transferred)));
     }
 
     /**
diff --git a/node_modules/fhenix-confidential-contracts/contracts/FHERC20UnshieldClaim.sol b/node_modules/fhenix-confidential-contracts/contracts/FHERC20UnshieldClaim.sol
index 05ed44c..92de519 100644
--- a/node_modules/fhenix-confidential-contracts/contracts/FHERC20UnshieldClaim.sol
+++ b/node_modules/fhenix-confidential-contracts/contracts/FHERC20UnshieldClaim.sol
@@ -30,15 +30,15 @@ abstract contract FHERC20UnshieldClaim {
     error LengthMismatch();
 
     function _createClaim(address to, uint64 value, euint64 claimable) internal {
-        _claims[euint64.unwrap(claimable)] = Claim({
-            ctHash: euint64.unwrap(claimable),
+        _claims[bytes32(euint64.unwrap(claimable))] = Claim({
+            ctHash: bytes32(euint64.unwrap(claimable)),
             requestedAmount: value,
             decryptedAmount: 0,
             decrypted: false,
             to: to,
             claimed: false
         });
-        _userClaims[to].add(euint64.unwrap(claimable));
+        _userClaims[to].add(bytes32(euint64.unwrap(claimable)));
     }
 
     function _handleClaim(
